sdk/src/Core/Amazon.Runtime/Internal/Auth/AbstractAWSSigner.cs

```diff
@@ -108,16 +108,8 @@ protected static string ComputeHash(byte[] data, string secretkey, SigningAlgori
                 throw new Amazon.Runtime.SignatureException("Failed to generate signature: " + e.Message, e);
             }
         }
-        public abstract void Sign(IRequest request, IClientConfig clientConfig, RequestMetrics metrics, string awsAccessKeyId, string awsSecretAccessKey);
 
-        public virtual void Sign(IRequest request, IClientConfig clientConfig, RequestMetrics metrics, BaseIdentity baseIdentity)
-        {
-            if (baseIdentity is AWSCredentials credentials)
-            {
-                var immutableCredentials = credentials.GetCredentials();
-                Sign(request, clientConfig, metrics, immutableCredentials?.AccessKey, immutableCredentials?.SecretKey);
-            }
-        }
+        public abstract void Sign(IRequest request, IClientConfig clientConfig, RequestMetrics metrics, BaseIdentity identity);
 
 #if AWS_ASYNC_API
         public virtual System.Threading.Tasks.Task SignAsync(
```

Comment:
nit: we can also rename the `baseIdentity` to `identity` similar to the sync version.

Reply:
Done